File: 백준/Gold/25332. 수들의 합 8/수들의 합 8.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.HashMap;
import java.util.Map;

public class Main {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int N = Integer.parseInt(br.readLine());
		int[] arr = Arrays.stream(br.readLine().split(" ")).mapToInt(Integer::parseInt).toArray();
		int[] brr = Arrays.stream(br.readLine().split(" ")).mapToInt(Integer::parseInt).toArray();

		int[] prefixA = new int[N + 1];
		int[] prefixB = new int[N + 1];
		for (int i=0; i<N; i++) {
			prefixA[i + 1] = prefixA[i] + arr[i];
			prefixB[i + 1] = prefixB[i] + brr[i];
		}

		int[] calAB = new int[N + 1];
		for (int i=1; i<=N; i++) {
			calAB[i] = prefixA[i] - prefixB[i];
		}

		long ans = 0;
		Map<Integer, Long> map = new HashMap<>();
		for (int i=0; i<=N; i++) {
			ans += map.getOrDefault(calAB[i], 0L);
			map.put(calAB[i], map.getOrDefault(calAB[i], 0L) + 1);
		}
		System.out.print(ans);
	}
}

File: 백준/Gold/25332. 수들의 합 8/수들의 합 8.py
import sys;input=sys.stdin.readline
n = int(input())
a = [0] + list(map(int,input().split()))
b = [0] + list(map(int,input().split()))
prefix_a = [0 for _ in range(n+1)]
prefix_b = [0 for _ in range(n+1)]
zero = [0 for _ in range(n+1)]
for i in range(1, n+1):
    prefix_a[i] = prefix_a[i-1] + a[i]
    prefix_b[i] = prefix_b[i-1] + b[i]

for i in range(1, n+1):
    zero[i] = prefix_b[i] - prefix_a[i]

cnt = {}
ans = 0

for i in range(n+1):
    ans += cnt.get(zero[i],0)
    cnt[zero[i]] = cnt.get(zero[i],0) + 1

print(ans)